packages/task/src/browser/task-configurations.ts

```diff
@@ -0,0 +1,257 @@
+            // one tasks.json file we're watching has changed?
+            if (this.hasWatchedFileChanged(changes.changes)) {
+                this.tasksMap.clear();
+                const promises: Promise<TaskOptions[]>[] = [];
+
+                // re-read all config files
+                for (const fileUri of this.watchedConfigFileUris) {
+                    promises.push(this.readTasks(fileUri));
+                }
+
+                Promise.all(promises)
+                    .then(tasksArrayArray => {
+                        for (const taskArray of tasksArrayArray) {
+                            for (const task of taskArray) {
+                                this.tasksMap.set(task.label, task);
+                            }
+                        }
+                        resolve();
+                    })
+                    .catch(err => {
+                        this.logger.error(err);
+                        reject(err);
+                    });
+            }
+        });
+    }
+
+    protected hasWatchedFileChanged(changes: FileChange[]): boolean {
+        for (const change of changes) {
+            for (const file of this.watchedConfigFileUris) {
+                if (change.uri === file) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    protected readTasks(uri: string): Promise<TaskOptions[]> {
+        return new Promise((resolve, reject) => {
+            this.fileSystem.exists(uri).then(exists => {
+                if (!exists) {
+                    reject(`tasks.json file does not exists: ${uri}`);
+                }
+                this.fileSystem.resolveContent(uri)
+                    .then(({ stat, content }) => {
+                        const strippedContent = jsoncparser.stripComments(content);
+                        const errors: ParseError[] = [];
+                        const tasks = jsoncparser.parse(strippedContent, errors);
+
+                        if (errors.length) {
+                            for (const e of errors) {
+                                this.logger.error(`Error parsing json at: ${uri}: ${e}`);
+                            }
+                            reject(`Error(s) parsing Tasks config file: ${uri}`);
+                        } else {
+                            resolve(this.filterDuplicates(tasks['tasks']));
+                        }
+                    })
+                    .catch(err => {
+                        reject(`Error(s) reading config file: ${uri} - see log for more details`);
+                    });
+            });
+        });
+    }
+
+    private filterDuplicates(tasks: TaskOptions[]): TaskOptions[] {
+        const filteredTasks: TaskOptions[] = [];
+        for (const task of tasks) {
+            if (filteredTasks.some(t => t.label === task.label)) {
+                this.logger.error(`Error parsing tasks.json: found duplicate entry for label ${task.label}`);
```

Comment:
This should ideally go into a marker for the editor. (maybe later)

Reply:
I have added a TODO note in the code about this.